lib/raxx/middleware/head.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Raxx.Middleware.Head do
+  @moduledoc """
+  An example router that allows you to handle HEAD requests with GET handlers
+  """
+
+  alias Raxx.Middleware
+  alias Raxx.Pipeline
+
+  @behaviour Middleware
+
+  @impl Middleware
+  def handle_head(request = %{method: :HEAD}, _config, pipeline) do
+    request = %{request | method: :HEAD}
+    state = :engage
+    {parts, pipeline} = Pipeline.handle_head(request, pipeline)
+
+    parts = modify_response_parts(parts, state)
+    {parts, state, pipeline}
+  end
+
+  def handle_head(request = %{method: _}, _config, pipeline) do
+    {parts, pipeline} = Pipeline.handle_head(request, pipeline)
+    {parts, :disengage, pipeline}
+  end
+
+  @impl Middleware
+  def handle_data(data, state, pipeline) do
```

Comment:
If you were too put the state and pipeline in a tuple you would be back to the `Raxx.Server` interface. :thinking: :thinking: 
```elixir
  @impl Server
  def handle_data(data, {state, pipeline}) do
    {parts, pipeline} = Pipeline.handle_data(data, pipeline)
    parts = modify_response_parts(parts, state)
    {parts, {state, pipeline}}
  end